File: gaia-sdk-java/gaia-sdk-mqtt/src/test/kotlin/gaia/sdk/mqtt/MqttSensorQueueTest.kt
/*
 * Copyright (c) 2016-2020, Leftshift One
 * __________________
 * [2020] Leftshift One
 * All Rights Reserved.
 * NOTICE:  All information contained herein is, and remains
 * the property of Leftshift One and its suppliers,
 * if any.  The intellectual and technical concepts contained
 * herein are proprietary to Leftshift One
 * and its suppliers and may be covered by Patents,
 * patents in process, and are protected by trade secret or copyright law.
 * Dissemination of this information or reproduction of this material
 * is strictly forbidden unless prior written permission is obtained
 * from Leftshift One.
 */

package gaia.sdk.mqtt

import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper
import gaia.sdk.mqtt.queue.ConvInteraction
import gaia.sdk.mqtt.queue.ConversationQueueType.INTERACTION
import gaia.sdk.mqtt.queue.QueueHeader
import gaia.sdk.spi.QueueOptions
import java.util.*
import kotlin.collections.HashMap

internal class MqttSensorQueueTest {

    private val objectMapper = ObjectMapper()

    // @Test
    fun test() {
        val queue = MqttSensorQueue(QueueOptions("mqtt.beta.gaia.leftshift.one", 443))
        val header = QueueHeader(UUID.fromString("4a87c137-3894-4580-ae20-8a4f621b75fd"), UUID.randomUUID())

        queue.connect()
                .andThen(queue.subscribe(INTERACTION, header) {
                    val type = object : TypeReference<ArrayList<HashMap<String, Any>>>() {}
                    val content = objectMapper.readValue(it.content, type)
                    println(content)
                })
                .andThen(queue.publishConvInteraction(header, ConvInteraction(mapOf("type" to "reception", "payload" to "", "attributes" to HashMap<String, Any>()))))
                .subscribe({}) {
                    it.printStackTrace()
                }

        Thread.sleep(100000)
    }

}

File: gaia-sdk-javascript/src/__tests__/MqttSensorQueue.test.ts
import {MqttSensorQueue, QueueHeader, QueueOptions} from '..';

describe.skip('MqttSensorQueue Test', () => {

    beforeEach(() => {
        jest.setTimeout(10000);
    });

    /**
     * This test will always be green, because we have no possibility to await the completion of publish.
     * TODO: Write a meaningful test once we can await the result of publish
     */
    it('connect', () => {
        const header = new QueueHeader('4a87c137-3894-4580-ae20-8a4f621b75fd', 'niceChannel');
        const mqttSensorQueue = new MqttSensorQueue(new QueueOptions('wss://mqtt.beta.gaia.leftshift.one/mqtt', 443, 'heimdall'));
        mqttSensorQueue.publishConvInteraction(header, {
            attributes: {},
            type: 'reception',
            payload: {}
        });
    });
});

